### DISCUSSION TITLE
Some way to show the output file in explorer

### DISCUSSION THREAD

**Discussion by rzmk** [dathere/qsv#1923] [Ideas]
Not sure how it would be implemented CLI-wise but I think it might be interesting to integrate this crate somehow: https://github.com/jf2048/showfile

For example if I have a `--output` file some way to automatically show that in file explorer when done.

**Comment by rzmk** [comment]
I'd suggest showing the file in an explorer is optional too since CI may not want such a feature to occur.

**Comment by jqnatividad** [comment]
I like it! And it nicely complements `prompt`...

Perhaps, if the `--output` parameter starts with a special reserved character like "!"?  For example:

`qsv stats data.csv -E --output !datastats.csv`

signals qsv to display datastats.csv in the current explorer for the platform...

WDYT?

**Reply by rzmk** [reply]
That sounds great. I’m curious how to help users figure that out aside from including it in the `—output` usage texts.

So long as file names can’t start with that special character that could work (could also maybe be modifiable with an env var but I’m not sure what system would allow ! to include that special character within the file path at the start).

**Reply by jqnatividad** [reply]
Mentioning it in the 100, the usage text, the example, and the walkthrough (which is overdue for an update), should be enough.

As a universally supported option suitewide, it should be quickly adopted.

And though "!" seems to be a technically legal character for filenames across platforms - in the real world, having it as a sentinel first character should be good enough for our purpose.

https://superuser.com/questions/358855/what-characters-are-safe-in-cross-platform-file-names-for-linux-windows-and-os

**Reply by rzmk** [reply]
Can an extension have an exclamation point at the end? I'm just thinking in case a user wants to use tab completions on an absolute/relative path before the file name it might stop the completions if prepended by an exclamation point (or could be time consuming to go all the way back to the start of a path to prepend it).

**Reply by jqnatividad** [reply]
Right, tab completions would be blocked by having "!" upfront...
Having it at the end actually may make it better as file extensions are really much more constrained and there seems to be no file extension with an exclamation point, never mind anything other than alphanumeric characters.

https://mimetype.io/all-types

**Comment by rzmk** [comment]
Also if the following issue is resolved then could also consider using the crate for any qsv commands that output to multiple files and/or if multiple `--output` is implemented: https://github.com/jf2048/showfile/issues/1
